aiida/orm/nodes/node.py

```diff
@@ -630,67 +630,187 @@ def extras_keys(self):
         """
         return self.backend_entity.extras_keys()
 
-    def list_objects(self, key=None):
+    def list_objects(self, path=None, key=None):
         """Return a list of the objects contained in this repository, optionally in the given sub directory.
 
+        .. deprecated:: 1.4.0
+            Keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.
+
+        :param path: the relative path of the object within the repository.
         :param key: fully qualified identifier for the object within the repository
         :return: a list of `File` named tuples representing the objects present in directory with the given key
         :raises FileNotFoundError: if the `path` does not exist in the repository of this node
         """
-        return self._repository.list_objects(key)
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn(
+                'keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.',
+                AiidaDeprecationWarning
+            )  # pylint: disable=no-member
+            path = key
+
+        return self._repository.list_objects(path)
 
-    def list_object_names(self, key=None):
+    def list_object_names(self, path=None, key=None):
         """Return a list of the object names contained in this repository, optionally in the given sub directory.
 
+        .. deprecated:: 1.4.0
+            Keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.
+
+        :param path: the relative path of the object within the repository.
         :param key: fully qualified identifier for the object within the repository
         :return: a list of `File` named tuples representing the objects present in directory with the given key
         """
-        return self._repository.list_object_names(key)
+        if key is not None:
+            if path is not None:
+                raise ValueError('cannot specify both `path` and `key`.')
+            warnings.warn(
+                'keyword `key` is deprecated and will be removed in `v2.0.0`. Use `path` instead.',
+                AiidaDeprecationWarning
+            )  # pylint: disable=no-member
+            path = key
 
-    def open(self, key, mode='r'):
+        return self._repository.list_object_names(path)
+
+    def open(self, path=None, mode='r', key=None):
         """Open a file handle to an object stored under the given key.
```

Comment:
Maybe `Open a file handle to an object stored under the given path.`? (or node local path or something like that, but changing the `key` reference to `path`)

Reply:
So, there are a couple of instances that you changed the docstrings to add the new param and deprecation warnings but the general description of the method still makes a reference to the old keywords. I didn't want to single all of them one by one in case maybe I'm missing something and these weren't supposed to be changed, but I can keep checking for this if you agree these need to be adapted as well.

Reply:
Thanks, I will go through the docstrings once more to correct those.